__tests__/components/Button.test.tsx

```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import { render } from '@testing-library/react';
+import { Button } from '@/components/Button';
+Button;
+
+describe('Button', () => {
+    it('should render', () => {
+        const handleOnClickMock = jest.fn();
+        const { container } = render(
+            <Button
+                type="button"
+                className="w-full md:w-auto bg-gray-200 px-4 md:px-8 py-3 hover:bg-gray-300 mt-2 md:mt-0 md:rounded-none"
+                onClick={handleOnClickMock}
+                children="generate"
+            />
+        );
+        expect(container).toMatchSnapshot();
+        expect(container).toHaveTextContent('generate');
```

Comment:
This snapshot testing generates an instance of the rendered component, each time you run the test and we don't need these snapshots while pushing the code. The better alternative will be 
```
   const { getByRole } = render(
            <Button type="submit"/>
        );

   const buttonElement = getByRole('button');
   expect(buttonElement).toHaveAttribute('type', 'submit');
```

Reply:
Done.